portal-impl/src/com/liferay/portlet/asset/service/impl/AssetCategoryServiceImpl.java

```diff
@@ -287,6 +287,14 @@ public AssetCategory getOrAddIncompleteCategory(
 			String externalReferenceCode, long groupId)
 		throws PortalException {
 
+		AssetCategory assetCategory =
+			assetCategoryLocalService.fetchAssetCategoryByExternalReferenceCode(
+				externalReferenceCode, groupId);
```

Comment:
I think we should use the fetchCategoryByExternalReferenceCode of the same (remote) service because otherwise we are not checking that you have the VIEW permission of the resource